shotover/src/server.rs

```diff
@@ -548,17 +551,37 @@ impl<C: CodecBuilder + 'static> Handler<C> {
 
         match transport {
             Transport::WebSocket => {
-                let ws_stream = tokio_tungstenite::accept_async(stream)
-                    .await
-                    .expect("Error during the websocket handshake occurred");
-
-                spawn_websocket_read_write_tasks(
-                    codec_builder,
-                    ws_stream,
-                    in_tx,
-                    out_rx,
-                    out_tx.clone(),
-                );
+                if let Some(tls) = &self.tls {
+                    let tls_stream = match tls.accept(stream).await {
+                        Ok(x) => x,
+                        Err(AcceptError::Disconnected) => return Ok(()),
+                        Err(AcceptError::Failure(err)) => return Err(err),
+                    };
+
+                    let ws_stream = tokio_tungstenite::accept_async(tls_stream)
```

Comment:
we could avoid duplicating this line like this:
![image](https://github.com/shotover/shotover-proxy/assets/5120858/6ed528b8-409c-431f-a271-9439671078e8)